### PR TITLE
Improve IDE support on symfony core code

### PR DESCRIPTION
**Pull request by e1himself**
- No code changes
- Absolutely BC
- Add return type hints comments
- Add class properties type hints comments
- Add method arguments type hints comments
- Fix inconsistency between doc comments and implementation (by fixing comments)

### CODE REVIEW

**Comment by e1himself** [comment]
Anybody here? :)

**Comment by j0k3r** [comment]
I'm here and afraid of checking all 107 changing files :fearful:

**Comment by e1himself** [comment]
:) @j0k3r those are just comments. 
The changes are just sane descriptions of actual method behavior. 
For most of them you can tell they're good just by looking at the diff.

Here are two (pretty straighforward) examples:

File: lib/command/sfCommandApplication.class.php [hunk 1]
```diff
@@ -18,19 +18,32 @@
...
+  /** @var sfTask */
+  protected $currentTask = null;
+  /** @var sfEventDispatcher */
+  protected $dispatcher = null;
+  /** @var array */
+  protected $options = array();
+  /** @var sfFormatter */
+  protected $formatter = null;
+  /** @var mixed */
+  protected $commandOptions;
```
Comment (GromNaN) [inline lib/command/sfCommandApplication.class.php]: Good catch. This attribute was not defined but dynamically assigned.

File: lib/controller/sfController.class.php [hunk 2]
```diff
@@ -20,17 +20,20 @@
...
-    $renderMode        = sfView::RENDER_CLIENT,
-    $maxForwards       = 5;
-
+  /** @var sfContext */
+  protected $context = null;
+  /** @var sfEventDispatcher */
+  protected $dispatcher = null;
+  /** @var string[] */
+  protected $controllerClasses = array();
+  protected $renderMode = sfView::RENDER_CLIENT;
```
Comment (GromNaN) [inline lib/controller/sfController.class.php]: No PHP doc for these ?

File: lib/form/addon/sfFormObject.class.php [hunk 3]
```diff
@@ -96,8 +96,8 @@ public function bindAndSave($taintedValues, $taintedFiles = null, $con = null)
 
     return false;
   }
-
-
+  
+
```
Comment (GromNaN) [inline lib/form/addon/sfFormObject.class.php]: Useless whitespaces added here.

File: lib/generator/sfModelGeneratorConfiguration.class.php [hunk 4]
```diff
@@ -464,6 +474,7 @@ public function getCredentials($action)
 
   public function getPager($model)
   {
+    // TODO: Probably `getPagerClass()` method should be abstract here. As well as `getPagerMaxPerPage`
```
Comment (GromNaN) [inline lib/generator/sfModelGeneratorConfiguration.class.php]: Sure, the method is used but never required.

File: lib/controller/sfController.class.php (hunk 2 above)
Comment (e1himself) [inline lib/controller/sfController.class.php]: Fixed.

File: lib/form/addon/sfFormObject.class.php (hunk 3 above)
Comment (e1himself) [inline lib/form/addon/sfFormObject.class.php]: Fixed with 99efd6e

**Comment by e1himself** [comment]
@GromNaN 

> @return void is not necessary, because implicit. And it seems to be non-standard.

My arguments for leaving it in are:

1. It's more obvious to change `@return something` => `@return void` to emphasize that it's returning nothing.
2. There's ton of `@return void` notations throughout symfony code.
3. It *is* in standard. It's just optional. See https://phpdoc.org/docs/latest/references/phpdoc/tags/return.html  
Also I think it will be even more commonly used after adopting PHP 7.1 with its `: void` return type hint.

----------------

Please check the changes and let me know if you think it's mergeable. Thanks :)

**Comment by e1himself** [comment]
Ping @GromNaN

**Comment by e1himself** [comment]
Ping @GromNaN @j0k3r

**Comment by e1himself** [comment]
Thanks :smile:

**Pull request merged**